src/sql/parts/connection/common/connectionManagementService.ts

```diff
@@ -1324,4 +1334,19 @@ export class ConnectionManagementService implements IConnectionManagementService
 		}
 		return Promise.reject('The given URI is not currently connected');
 	}
+
+	public getTabColorForUri(uri: string): string {
+		if (!WorkbenchUtils.getSqlConfigValue<string>(this._workspaceConfigurationService, 'enableTabColors')) {
+			return undefined;
+		}
+		let connectionProfile = this.getConnectionProfile(uri);
+		if (!connectionProfile) {
+			return undefined;
+		}
+		let matchingGroup = this._connectionStore.getGroupFromId(connectionProfile.groupId);
```

Comment:
what do we do about connection in the root directory?  presumably they draw with the default tab color?

Reply:
Yes, those just get drawn with the default color. They'll look just like the "User Settings" tab in my screenshot